src/test/scala/edu/ie3/simona/agent/em/EmAgentIT.scala

```diff
@@ -0,0 +1,433 @@
+            emResult.getP should equalWithTolerance(
+              (-0.000566087824).asMegaWatt
+            )
+            emResult.getQ should equalWithTolerance(0.001073120041.asMegaVar)
+        }
+
+        scheduler.expectMessage(Completion(emAgentActivation, Some(7200)))
+
+        /* TICK 7200
+         LOAD: 0.000269 MW (unchanged)
+         PV:  -0.003797 MW
+         Heat pump: running (turned on from last request), can be turned on and stay off
+         -> set point ~3.5 kW (bigger than 50 % rated apparent power): stays turned on with unchanged state
+         -> remaining 0 MW
+         */
+
+        emAgentActivation ! Activation(7200)
+
+        weatherDependentAgents.foreach {
+          _ ! ProvideWeatherMessage(
+            7200,
+            weatherService.ref.toClassic,
+            WeatherData(
+              WattsPerSquareMeter(300d),
+              WattsPerSquareMeter(500d),
+              Celsius(0d),
+              MetersPerSecond(0d),
+            ),
+            Some(14400),
+          )
+        }
+
+        resultListener.expectMessageType[ParticipantResultEvent] match {
+          case ParticipantResultEvent(emResult: EmResult) =>
+            emResult.getInputModel shouldBe emInput.getUuid
+            emResult.getTime shouldBe 7200.toDateTime
+            emResult.getP should equalWithTolerance(0.00132184544484.asMegaWatt)
+            emResult.getQ should equalWithTolerance(0.001073120041.asMegaVar)
+        }
+
+        scheduler.expectMessage(Completion(emAgentActivation, Some(14400)))
+
+        /* TICK 14400
+         LOAD: 0.000269 MW (unchanged)
+         PV:  -0.000066 MW
+         Heat pump: Is not running, can still run or be turned off
+         -> flex signal is 0 MW: Heat pump is turned off
+         */
+
+        emAgentActivation ! Activation(14400)
+
+        // it got cloudy now...
+        weatherDependentAgents.foreach {
+          _ ! ProvideWeatherMessage(
+            14400,
+            weatherService.ref.toClassic,
+            WeatherData(
+              WattsPerSquareMeter(5d),
+              WattsPerSquareMeter(5d),
+              Celsius(0d),
+              MetersPerSecond(0d),
+            ),
+            Some(21600),
+          )
+        }
+
+        resultListener.expectMessageType[ParticipantResultEvent] match {
+          case ParticipantResultEvent(emResult: EmResult) =>
+            emResult.getInputModel shouldBe emInput.getUuid
+            emResult.getTime shouldBe 14400L.toDateTime
+            emResult.getP should equalWithTolerance(0.000202956264.asMegaWatt)
+            emResult.getQ should equalWithTolerance(0.000088285537.asMegaVar)
+        }
+
+        scheduler.expectMessage(Completion(emAgentActivation, Some(21600)))
+
+        /* TICK 21600
+         LOAD: 0.000269 MW (unchanged)
+         PV:  -0.000032 MW
+         Heat pump: Is not running, can run or be turned off
+         -> flex signal is 0 MW: Heat pump is turned off
```

Comment:
same here, stays off?

Reply:
No, because it had been running up to the current tick

Reply:
but then it's
```suggestion
         Heat pump: Is running, can run or be turned off
         -> flex signal is 0 MW: Heat pump is turned off
```

Reply:
Improved the phrasing a bit...